src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultAntiforgeryTokenSerializer.cs

```diff
@@ -156,7 +161,14 @@ public string Serialize(AntiforgeryToken token)
 
                 writer.Flush();
                 var stream = serializationContext.Stream;
-                return WebEncoders.Base64UrlEncode(_cryptoSystem.Protect(stream.ToArray()));
+                var bytes = _cryptoSystem.Protect(stream.ToArray());
```

Comment:
The way to do this would be to:
1). Use `new MemoryString(byte[])` 
2). Implement a custom memory stream based on pooled memory